.github/workflows/ci.yml

```
@@ -31,6 +31,9 @@ jobs:
       - name: Run tests
         run: poetry run pytest --cov-report=xml
 
+      - name: Test building the docs
+        run: poetry run mkdocs build --strict
```

I'd do things in this order:

```suggestion
      - name: Test building the docs
        run: poetry run mkdocs build --strict

      - name: Run tests
        if: always()
        run: poetry run pytest --cov-report=xml
```

As it stands, it'll run the tests and, if that fails, it won't check the docs. Also, if there are doc warnings, it won't upload coverage.

---

Agree with this suggestion though GitHub suggests using `${{ !cancelled() }}` rather than `always()`. https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#always

---

Done, with Chris alternative.